src/aws-fsx/ontap-file-system.ts

```diff
@@ -0,0 +1,514 @@
+    }
+    if (
+      haPairs > 1 &&
+      [OntapDeploymentType.SINGLE_AZ_1, OntapDeploymentType.MULTI_AZ_1, OntapDeploymentType.MULTI_AZ_2].includes(
+        deploymentType,
+      )
+    ) {
+      throw new Error(`\'haPairs\' must be 1 for deployment type ${deploymentType}`);
+    }
+  }
+
+  private validateSubnets(
+    deploymentType: OntapDeploymentType,
+    vpcSubnets: aws_ec2.ISubnet[],
+    preferredSubnet?: aws_ec2.ISubnet,
+  ): void {
+    if (
+      (deploymentType === OntapDeploymentType.MULTI_AZ_1 || deploymentType === OntapDeploymentType.MULTI_AZ_2) &&
+      !preferredSubnet
+    ) {
+      throw new Error("'preferredSubnet' must be specified for multi-AZ file systems");
+    }
+    if (
+      (deploymentType === OntapDeploymentType.SINGLE_AZ_1 || deploymentType === OntapDeploymentType.SINGLE_AZ_2) &&
+      preferredSubnet
+    ) {
+      throw new Error("'preferredSubnet' can only be specified for multi-AZ file systems");
+    }
+    if (preferredSubnet && !vpcSubnets.includes(preferredSubnet)) {
+      throw new Error("'preferredSubnet' must be one of the specified 'vpcSubnets'");
+    }
+  }
+
+  private validateRouteTables(deploymentType: OntapDeploymentType, routeTables?: aws_ec2.IRouteTable[]): void {
+    if (routeTables == null || routeTables.length === 0) {
+      return;
+    }
+    if (deploymentType !== OntapDeploymentType.MULTI_AZ_1 && deploymentType !== OntapDeploymentType.MULTI_AZ_2) {
+      throw new Error("'routeTables' can only be specified for multi-AZ file systems");
+    }
+  }
+
+  private validateThroughputCapacity(
+    deploymentType: OntapDeploymentType,
+    throughputCapacity?: number,
+    throughputCapacityPerHaPair?: number,
+    haPair: number = 1,
+  ): void {
+    if (
+      Token.isUnresolved(throughputCapacity) ||
+      Token.isUnresolved(throughputCapacityPerHaPair) ||
+      Token.isUnresolved(haPair)
+    ) {
+      return;
+    }
+    if (throughputCapacity == null && throughputCapacityPerHaPair == null) {
+      return;
+    }
+    if (throughputCapacity != null && throughputCapacityPerHaPair != null) {
+      throw new Error("'throughputCapacity' and 'throughputCapacityPerHaPair' cannot be specified at the same time");
+    }
+    if (haPair <= 0 || !Number.isInteger(haPair)) {
+      throw new Error(`'haPair' must be a positive integer, got ${haPair}`);
+    }
+
+    // Calculate the throughput per HaPair and use it for validation,
+    // regardless of whether `throughputCapacity` or `throughputCapacityPerHaPair` is defined.
+    const throughputPerHaPair = throughputCapacityPerHaPair ?? throughputCapacity! / haPair;
+
+    const validValues: { [key in OntapDeploymentType]: number[] } = {
+      SINGLE_AZ_1: [128, 256, 512, 1024, 2048, 4096],
+      MULTI_AZ_1: [128, 256, 512, 1024, 2048, 4096],
+      SINGLE_AZ_2: [1536, 3072, 6144],
+      MULTI_AZ_2: [384, 768, 1536, 3072, 6144],
+    };
```

Comment:
These values are described in cfn docs. 

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-fsx-filesystem-ontapconfiguration.html#cfn-fsx-filesystem-ontapconfiguration-throughputcapacityperhapair